src/cdk/observers/observe-content.ts

```diff
@@ -35,6 +33,56 @@ export class MutationObserverFactory {
   }
 }
 
+
+/** A factory that creates ContentObservers. */
+@Injectable({providedIn: 'root'})
+export class ContentObserverFactory {
+  constructor(private _mutationObserverFactory: MutationObserverFactory, private _ngZone: NgZone) {}
+
+  create(element: Element, debounce?: number) {
+    const changes = new Subject<MutationRecord[]>();
+    const observer = this._ngZone.runOutsideAngular(
+        () => this._mutationObserverFactory.create((mutations) => changes.next(mutations)));
+    return new ContentObserver(element, observer, changes, debounce);
+  }
+}
+
+
+/** A class that observes an element for content changes. */
+export class ContentObserver {
```

Comment:
We could simplify the whole approach a bit by not having this class to begin with. Instead, `ContentObserverFactory.observe` can return an observable which creates a `MutationObserver` when something subscribes to it and then tears down the observer when there are no more subscriptions. [We did something similar in the `ScrollDispatcher`](https://github.com/angular/material2/blob/master/src/cdk/scrolling/scroll-dispatcher.ts#L84). It means that we lose the ability to pause an observer and resume it with the same `MutationObserver`, but AFAIK there isn't much overhead to creating mutation observers.

Reply:
I propose keeping `ContentObserverFactory` how it was before (docs private API that we use internally), and then making `ContentObserver` an injectable that works like `ScrollDispatcher` and `BreakpointObserver`; I think having the user need to interact with two different types is more work than it has to be